src/mobster/oci/cosign.py

```diff
@@ -96,15 +111,29 @@ async def fetch_sbom(self, image: Image) -> SBOM:
 
         Args:
             image (Image): Image to fetch the SBOM of.
+                This image MUST have a resolved hash!
         """
-        with make_oci_auth_file(image.reference) as authfile:
-            code, stdout, stderr = await run_async_subprocess(
-                ["cosign", "download", "sbom", image.reference],
-                env={"DOCKER_CONFIG": str(authfile.parent)},
-                retry_times=3,
-            )
-
-        if code != 0:
-            raise SBOMError(f"Failed to fetch SBOM {image}: {stderr.decode()}")
-
-        return SBOM.from_cosign_output(stdout, image.reference)
+        assert image.digest, f"Cannot fetch SBOM for an image without a digest: {image}"
+        provenance = await self.fetch_latest_provenance(image)
+        sbom_url = provenance.get_sbom_blob_url(image)
+        with tempfile.NamedTemporaryFile() as tmp_file:
+            oras_command = [
```

Comment:
Don't you need to prepare the auth file and provide it to the oras? I'll let @jedinym comment on that since he faced couple of authfile issues in past.

Reply:
Actually yes, `oras` has a specific flag for the docker config. I will use that instead of the variable.